core/src/main/java/smile/classification/LogisticRegression.java

```diff
@@ -823,7 +834,78 @@ public double f(double[] w, double[] g) {
         }
     }
 
-    /**
+	@Override
+	public void learn(double[] x, int y) {
+		if (y < 0 || y >= k) {
+			throw new IllegalArgumentException("Invalid label");
+		}
+
+		if (x.length != p) {
+			throw new IllegalArgumentException("Invalid input vector size: " + x.length);
+		}
+
+		if (k == 2) {
+			// calculate gradient for incoming data
+			double wx = dot(x, w);
+			double res = y - Math.logistic(wx);
+
+			double[] g = new double[p + 1];
```

Comment:
no need to create a new an array here. It introduces memory fragment and slows down the system when we update the model with a lot of samples. It is easy to combine this loop with the below one and just use a local variable.